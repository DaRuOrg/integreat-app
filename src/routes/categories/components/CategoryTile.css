@import url('../../../modules/common/constants/helpers.css');

.category-tiles {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-evenly;
  padding: 10px 0;
}

.category {
  min-width: 25%; /* max 4 columns per row */
  margin-bottom: 20px;
}

.category > a {
  @mixin remove-a;
  display: block;
  width: 160px;
  max-width: 35vw;
  margin: 0 auto;
  padding: 0 10px;
}

.category-thumbnail {
  position: relative;
  display: block;
  width: 100%;
  margin: 0 auto;
  padding-top: 100%;
}

.category-thumbnail img {
  position: absolute;
  top: 0;
  width: 100%;
  height: 100%;
  transition: transform 0.2s;
  object-fit: contain;
}

.category-thumbnail img:hover {
  transform: scale(1.01);
}

.category-title {
  margin: 5px 0;
  text-align: center;
}
