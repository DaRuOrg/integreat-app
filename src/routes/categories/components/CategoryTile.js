import React from 'react'
import PropTypes from 'prop-types'
import { Link } from 'redux-little-router'

import CategoryModel from 'modules/endpoint/models/CategoryModel'

import style from './CategoryTile.css'

/**
 * Displays a single CategoryTile
 */
class CategoryTile extends React.Component {
  static propTypes = {
    category: PropTypes.instanceOf(CategoryModel).isRequired
  }

  render () {
    return (
      <div className={style.category}>
        <Link href={this.props.category.url}>
          <div className={style.categoryThumbnail}><img src={this.props.category.thumbnail} /></div>
          <div className={style.categoryTitle}>{this.props.category.title}</div>
        </Link>
      </div>
    )
  }
}

export default CategoryTile
