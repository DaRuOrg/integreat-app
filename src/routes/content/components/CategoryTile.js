import React from 'react'
import PropTypes from 'prop-types'
import { Col } from 'react-flexbox-grid'
import { Link } from 'redux-little-router'
import style from './CategoryTile.css'
import PageModel from 'modules/endpoint/models/PageModel'

class CategoryTile extends React.Component {
  static propTypes = {
    page: PropTypes.instanceOf(PageModel).isRequired,
    url: PropTypes.string.isRequired
  }

  render () {
    return (
      <Col xs={6} sm={4} className={style.category}>
        <Link href={this.props.url}>
          <img className={style.categoryThumbnail} src={this.props.page.thumbnail}/>
          <div className={style.categoryTitle}>{this.props.page.title}</div>
        </Link>
      </Col>
    )
  }
}

export default CategoryTile
