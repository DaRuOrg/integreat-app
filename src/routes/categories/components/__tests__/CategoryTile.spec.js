import React from 'react'
import { shallow } from 'enzyme'

import CategoryTile from '../CategoryTile'

const category = {
  id: 3649,
  url: '/augsburg/de/willkommen',
  title: 'Willkommen',
  content: 'this is a test content',
  parentId: 0,
  parentUrl: '/augsburg/de',
  order: 11,
  availableLanguages: {
    en: 4804, ar: 4819, fa: 4827
  },
  thumbnail: 'https://cms.integreat-apâ€¦03/Beratung-150x150.png'
}

describe('CategoryTile', () => {
  it('should render and match snapshot', () => {
    const wrapper = shallow(
      <CategoryTile category={category} />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
