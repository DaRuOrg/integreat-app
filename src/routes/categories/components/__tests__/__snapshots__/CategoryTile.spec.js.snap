// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryTile should render and match snapshot 1`] = `
<Col
  className="category"
  md={3}
  sm={4}
  xs={6}
>
  <Connect(Link)
    href="/augsburg/de/willkommen"
  >
    <div
      className="thumbnailSizer"
    >
      <div
        className="categoryThumbnail"
      >
        <img
          src="https://cms.integreat-apâ€¦03/Beratung-150x150.png"
        />
      </div>
    </div>
    <div
      className="categoryTitle"
    >
      Willkommen
    </div>
  </Connect(Link)>
</Col>
`;
